src/client/components/stage-details/stage-details.component.js

```diff
@@ -0,0 +1,171 @@
+
+  // Select the last stage to display as default
+  const [selectedStageNumber, setSelectedStageNumber] = useState(
+    Math.max(...stages.map((s) => s.id)),
+  );
+
+  // Get data for the currently selected stage
+  const [dataForStage, setDataForStage] = useState([]);
+
+  useEffect(() => {
+    setDataForStage(
+      cropData
+        ? cropData.filter(
+            (dataPoint) =>
+              parseInt(dataPoint.fk_crop_stage_id, 10) === selectedStageNumber,
+          )
+        : [],
+    );
+  }, [selectedStageNumber, cropData]);
+
+  function handleInputChange(e, level, parameter) {
+    const valueEntered = e.target.value;
+    e.preventDefault();
+    setCropData((prev) => {
+      return prev.map((dataPoint) => {
+        if (
+          Number(dataPoint.fk_crop_stage_id) === selectedStageNumber &&
+          dataPoint.parameter === parameter
+        ) {
+          const updatedDataPoint = {
+            ...dataPoint,
+            [`${level}_value`]: valueEntered,
+          };
+          return updatedDataPoint;
+        }
+        return dataPoint;
+      });
+    });
+  }
+
+  function saveUpdatedCropData(e) {
+    e.preventDefault();
+    /* TO DO: SAVE UPDATED cropData */
+  }
+
+  return (
+    <div className="stage-details-container">
+      {/* Section header */}
+      <SectionHeader onClick={() => null} tabIndex={0}>
+        Stage Details
+      </SectionHeader>
+
+      <form
+        className="stage-values-form"
+        onSubmit={(e) => saveUpdatedCropData(e)}
+      >
+        {/* Toggle buttons to select stage */}
+        <div className="stage-buttons-container">
+          {stages.map((stage) => {
+            return (
+              <Button
+                key={stage.id}
+                type="toggle"
+                onClick={(e) => {
+                  e.preventDefault();
+                  setSelectedStageNumber(stage.id);
+                }}
+                toggled={selectedStageNumber === stage.id}
+              >
+                {stage.name}
+              </Button>
+            );
+          })}
+        </div>
+
+        {/* Column containing parameter labels */}
+        <div className="stage-values-container">
+          <div className="title-column">
+            {Object.keys(parametersAndLabels).map((name) => (
+              <p className="stage-details-title-item" key={name}>
+                {parametersAndLabels[name]}
+              </p>
+            ))}
+          </div>
+
+          {/* Columns containing values per parameter and level */}
+          {levels.map((level) => {
+            return (
+              <div
+                className={`${level}-column`}
+                key={`stage-${selectedStageNumber}-${level}-column`}
+              >
+                {Object.keys(parametersAndLabels).map((parameter) => {
+                  const dataForParameter = dataForStage.filter((dataPoint) => {
+                    return dataPoint.parameter === parameter;
+                  });
```

Comment:
If you only want the first and one item in the array you might as well use `.find()` rather than `.filter()`, then you can avoid the `[0]`

Reply:
Ty! ✔️